anaDACScan.py

```diff
@@ -0,0 +1,142 @@
+
+from gempython.gemplotting.mapping.chamberInfo import chamber_config
+
+import argparse
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser(description='Arguments to supply to anaDACScan.py')
+
+    parser.add_argument('infilename', type=str, help="Filename from which input information is read", metavar='infilename')
+    parser.add_argument("calFileFile", type=str, help="File specifying which calFile to use for each OH. Format: OH1 filenameOH1 <newline> OH2 filenameOH2 <newline> etc", metavar="calFileFile")
+    parser.add_argument('-o','--outfilename', dest='outfilename', type=str, default="DACFitData.root", help="Filename to which output information is written", metavar='outfilename')
+    parser.add_argument('--assignXErrors', dest='assignXErrors', action='store_true', help="Whether to assign errors for the X variable (which is actually plotted on the y-axis)")
+
+    args = parser.parse_args()
+
+    print("Analyzing: '%s'"%args.infilename)
+
+    # Set default histogram behavior
+    r.TH1.SetDefaultSumw2(False)
+    r.gROOT.SetBatch(True)
+    r.gStyle.SetOptStat(1111111)
+
+    dacScanFile = r.TFile(args.infilename)
+
+    import numpy as np
+    import root_numpy as rp
+    
+    # Determine which links present
+    list_bNames = ['link']
+    ohArray = rp.tree2array(tree=dacScanFile.dacScanTree, branches=list_bNames)
+    ohArray = np.unique(ohArray['link'])
+
+    # Determine which DAC was scanned and against which ADC
+    nameX = ""
+    nameY = ""
+    for event in dacScanFile.dacScanTree:
+        nameX = event.nameX.data()
+        nameY = event.nameY.data()
+        break # all entries will be the same
+
+    if nameY not in ['ADC0', 'ADC1']:
+        print("Error: unexpected value of nameY: '%s'"%nameY)
+        exit(1)
+
+    calInfo = {}
+
+    for line in open(args.calFileFile):
+        line = line.strip(' ').strip('\n')
+        first = line.split(' ')[0].strip(' ')
+        second = line.split(' ')[1].strip(' ')
+        tuple_calInfo = parseCalFile(second)
+        calInfo[int(first)] = {'slope' : tuple_calInfo[0], 'intercept' : tuple_calInfo[1]}
+        
+    # Initialize nested dictionaries
+    dict_dacScanResults = nesteddict()
+    dict_dacScanFuncs = nesteddict()
+
+    # Initialize a TGraphErrors for each vfat
+    for link in ohArray:
+        for vfat in range(0,24):
+             dict_dacScanResults[link][vfat] = r.TGraphErrors()
+
+    outputFiles = {}         
+             
+    for link in ohArray:         
+        outputFiles[link] = r.TFile("$DATA_PATH/"+chamber_config[link]+"/dacScans/scandate/"+args.outfilename,'recreate')
+             
+    # Loop over events in the tree and fill plots
+    for event in dacScanFile.dacScanTree:
+        link = event.link
+        vfat = event.vfatN
+
+        if link not in calInfo.keys():
+            print('Error: calibration file for link %i was not provided'%link)
+            exit(1)
```

Comment:
Why `exit()`?  The other links may still be valid.  I would add either a `raw_input` prompt for the user to provide a calibration file for this link (adding it to `calInfo`), or allow the user to state "ignore" in which this link is always skipped.  If this is too challenging then a simple `continue` instead of `exit()`.

Reply:
This logic is unnecessary now, since I now remove OHs without a calFile from the OH array.